### DISCUSSION TITLE
Best case strategy

### DISCUSSION THREAD

**Discussion by RezaKakooee** [ContinualAI/avalanche#666] [Q&A] [answered]
Hi, I am new to using Avalanche but already became a bit familiar with it. I have a question regarding the supported CL strategies. Does Avalanche support a re-training strategy by default? Let's assume we have a class-incremental-learning case. By re-training, I mean first, we train the model for the initial classes, and then we have access to the initial dataset and the new class. So, we combine all the data and re-train the entire model again. It should be the best case strategy in terms of accuracy, but of course, not in terms of computation/data privacy and so on. But, anyway, I would like to know if such a strategy is supported by default or not? Many thanks in advance.

**Comment by AndreaCossu** [comment, marked as answer]
I think you are looking for the [Cumulative strategy](https://github.com/ContinualAI/avalanche/blob/master/avalanche/training/strategies/cumulative.py)

**Reply by RezaKakooee** [reply]
Hi, many thanks for your quick reply. Yes, exactly. I also just found [joint_train](https://github.com/ContinualAI/avalanche/blob/68290d32f9710639d5fda6d17da4072a4458f702/examples/joint_training.py)

It seems that `cumulative` and `joint_train` are the same. Right?

**Reply by AndreaCossu** [reply]
No, joint training trains the model once on the concatenation of all the experiences. On split mnist it is like training on the entire MNIST once.

**Reply by RezaKakooee** [reply]
So, but `cumulative` first train on the dataset of the 1st task, and then concatenate 1st and 2nd datasets, and next re-train the entire model on the concatenated dataset?

**Reply by AndreaCossu** [reply]
exactly

**Reply by RezaKakooee** [reply]
Got it! Many thanks :)
